beacon_chain/networking/eth2_network.nim

```diff
@@ -1295,9 +1336,35 @@ proc checkPeer(node: Eth2Node, peerAddr: PeerAddr): bool =
       trace "Recently connected"
       false
     else:
-      true
+      # check if the normal switch is already using the same connection
+      # avoid any overlap with normal peer use to prevent leakage or correlation
+      if node.torpeerPool.hasPeer(peerId) : 
```

Comment:
I'm not sure this is a good idea, this could actually cause a correlation

Reply:
The point is that both normal switch and tor switch are looking for attestation topic subnet peers! So we should have a way to avoid overlap as the attacker can build a correlation of active normal connections from peer with the concurrent connections through tor within the same time period! So that's why torswitch will check if normal is already connected with a peer that it should avoid making connection to and viceversa.

Ah, there could be a better way. Why not just keep attesstion_net peers specifc for torswitchh only. So the normal switch should not find attestation subnet peers and make connections to them all!